reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -618,4 +661,284 @@ public void shutdown() {}
+                + " availability: " + availability()
+                + "]->" + child.toString();
+        }
+
+        /**
+         * Subscriber wrapper used for request/response interaction model, measure and collect
+         * latency information.
+         */
+        private class LatencySubscriber<U> implements Subscriber<U> {
+            private final Subscriber<U> child;
+            private final WeightedSocket socket;
+            private final AtomicBoolean done;
+            private long start;
+
+            LatencySubscriber(Subscriber<U> child, WeightedSocket socket) {
+                this.child = child;
+                this.socket = socket;
+                this.done = new AtomicBoolean(false);
+            }
+
+            @Override
+            public void onSubscribe(Subscription s) {
+                start = incr();
+                child.onSubscribe(new Subscription() {
+                    @Override
+                    public void request(long n) {
+                        s.request(n);
+                    }
+
+                    @Override
+                    public void cancel() {
+                        if (done.compareAndSet(false, true)) {
+                            s.cancel();
+                            decr(start);
+                        }
+                    }
+                });
+            }
+
+            @Override
+            public void onNext(U u) {
+                child.onNext(u);
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                if (done.compareAndSet(false, true)) {
+                    child.onError(t);
+                    long now = decr(start);
+                    if (t instanceof TransportException || t instanceof ClosedChannelException) {
+                        removeSocket(socket);
+                    } else if (t instanceof TimeoutException) {
+                        observe(now - start);
+                    }
+                }
+            }
+
+            @Override
+            public void onComplete() {
+                if (done.compareAndSet(false, true)) {
+                    long now = decr(start);
+                    observe(now - start);
+                    child.onComplete();
+                }
+            }
+        }
+
+        /**
+         * Subscriber wrapper used for stream like interaction model, it only counts the number of
+         * active streams
+         */
+        private class CountingSubscriber<U> implements Subscriber<U> {
+            private final Subscriber<U> child;
+            private final WeightedSocket socket;
+
+            CountingSubscriber(Subscriber<U> child, WeightedSocket socket) {
+                this.child = child;
+                this.socket = socket;
+            }
+
+            @Override
+            public void onSubscribe(Subscription s) {
+                pendingStreams.incrementAndGet();
```

Comment:
It will be better, if you access `WeightedSocket` member variables using `socket.*` instead of direct access. It is easier to understand.